### PR TITLE
Fix issue with infinite loop

### LINKED ISSUE #11: Infinite loop for input values that do not require sanitization
There appears to be an issue in which inputs whose values do not require sanitization invoke the action being passed into one-way-input. This causes an infinite loop, as the value for the input is always changing as the action fires in each loop.

For example, inside of `_processNewValue()`, the default value of `_sanitizedValue` is `undefined` while `sanitizeInput` returns `false`:

```js
  _processNewValue(rawValue) {
    const value = this.sanitizeInput(rawValue);

    if (this._sanitizedValue !== value) {
      this._sanitizedValue = value;
      this.attrs.update(value);
    }
  }
```

### LINKED ISSUE #12: Action thrown on initial render?
Given the action is named `update`, and initial render doesn't change the value, I don't think it should trigger on initial render. Was this done on purpose?

### PR DESCRIPTION
**Pull request by poteto**
- Remove the new Error thrown when `update` is not passed, as user can
  now pass in other types of actions like `onenter` (responsibility of
  the user to implement an action for the new value)
- Update readme
- Fix issue where user would need to specify both `checked` and `value` for checkboxes

Closes #11, closes #10
Should also close #12, as the action will now only fire if `_sanitizedValue !== rawValue` (set in `didInitAttrs`)

### CODE REVIEW

**Comment by poteto** [comment]
cc @rwjblue

File: addon/components/one-way-input.js [hunk 1]
```diff
@@ -86,13 +86,12 @@ export default Component.extend({
     return input;
   },
 
+  didInitAttrs() {
```
Comment (rwjblue) [inline addon/components/one-way-input.js]: Can you tweak this to use `init` instead? We will likely be deprecating `didInitAttrs` (it provides roughly no value over just using `init`) in the future.

Comment (poteto) [inline addon/components/one-way-input.js]: Yes sir!

File: addon/components/one-way-input.js [hunk 2]
```diff
@@ -47,9 +47,10 @@ export default Component.extend({
   keyUp(event) { this._interpretKeyEvents(event); },
 
   _interpretKeyEvents(event) {
-    const methodName = this.KEY_EVENTS[event.keyCode];
+    let methodName = this.KEY_EVENTS[event.keyCode];
 
     if (methodName) {
+      this._sanitizedValue = null;
```
Comment (rwjblue) [inline addon/components/one-way-input.js]: Why does `this._sanitizedValue` get set to `null` here?

Comment (poteto) [inline addon/components/one-way-input.js]: It might be a bit of a hack, but it is needed so key events can fire. Otherwise, the `_sanitizedValue` is always going to be up to date with the raw value at time of key up

Comment (rwjblue) [inline addon/components/one-way-input.js]: Does the test added here demonstrate that?  I'm OK with this as long as the edge cases you mentioned are under test...

Comment (poteto) [inline addon/components/one-way-input.js]: Yup, I split out the acceptance test for key events with a separate input. That input has no `update` action, only an `onenter` action.

Comment (rwjblue) [inline addon/components/one-way-input.js]: Perfect, thank you!

**Comment by poteto** [comment]
@homu r+
